torch/_decomp/decompositions.py

```diff
@@ -3248,6 +3248,76 @@ def upsample_bilinear2d(
     return result
 
 
+@register_decomposition(aten.replication_pad2d.default)
+@pw_cast_for_opmath
+def replication_pad2d(input: Tensor, padding: List[int]) -> Tensor:
+    result_size = list(input.size())
+    padding_left = padding[0]
+    padding_right = padding[1]
+    padding_top = padding[2]
+    padding_bottom = padding[3]
+    result_size[-1] += padding_left + padding_right
+    result_size[-2] += padding_top + padding_bottom
+    result = torch.empty(result_size, device=input.device, dtype=input.dtype)
+
+    if padding_left < 0:
+        input = input[..., -padding_left:]
+        padding_left = 0
+
+    if padding_right < 0:
+        input = input[..., :padding_right]
+        padding_right = 0
+
+    if padding_top < 0:
+        input = input[..., -padding_top:, :]
+        padding_top = 0
+
+    if padding_bottom < 0:
+        input = input[..., :padding_bottom, :]
+        padding_bottom = 0
+
+    # Middle
+    result[
+        ...,
+        padding_top : padding_top + input.size(-2),
+        padding_left : padding_left + input.size(-1),
+    ] = input
```

Comment:
I tried running `torch.compile` on `torch.ops.aten.replication_pad2d` and it throws an error, apparently because it doesn't like all these index puts. But then I realized there's a significantly simpler way to implement this using `torch.cat` instead of index puts. `torch.compile` seems to work fine with that. I'll  push an update soon